node/node_handler.go

```diff
@@ -306,6 +301,12 @@ func (node *Node) PostConsensusProcessing(newBlock *types.Block) {
 			node.ConfirmedBlockChannel <- newBlock
 		}()
 	}
+
+	if node.NodeConfig.Role() == nodeconfig.BeaconLeader {
+		utils.GetLogInstance().Info("Updating staking list")
+		node.UpdateStakingList(node.QueryStakeInfo())
+		node.printStakingList()
+	}
 }
```

Comment:
Does it make logical difference that `UpdateStakingList` should happen after Broadcast of new block or after DRG (in case it is epoch block).

Reply:
If not, then this file doesn't need to be checked-in

Reply:
UpdateStakingList should happen after the block is added to the chain.